pkg/apis/testing/v1alpha1/testdefinition_types.go

```diff
@@ -0,0 +1,74 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TestDefinitionSpec defines the desired state of TestDefinition
+type TestDefinitionSpec struct {
+	PodSpec v1.PodSpec `json:"pod_spec"`
+	// If there are some problems with given test, we add possibility to don't execute them
+	// On Testsuite level such test should be marked as a skipped
+	Skip bool `json:"skip,omitempty"`
+	// If test is working on data that can be modified by another test,
+	// I would like to run it in separation.
+	DisableConcurrency bool `json:"disable_concurrency,omitempty"`
+	// List of components this test depends on.
+	Components []string `json:"components,omitempty"`
```

Comment:
I am also not convinced to the `components` because it can be treated as "Kyma components". Here we are going to provide list of test dependencies. If my test bind to a Function instance of Redis, this list should contain:
- service-catalog
- service-catalog-addons
- serverless/kubeless
- helm-broker 

Tags are too generic IMO. Maybe:
- Dependencies
- test_objects (https://en.wikipedia.org/wiki/System_under_test)

?

Reply:
Now what possibly can be a dependency and how can we find out if they are present? First of all we would probably require Deployments to be present. Could there be any other kind. We also need to specify namespace where to expect that dependency as application-connector for example runs in different namespace. 

How about following `RoleBinding`:
```
subjects:
- kind: User
  name: jane # Name is case sensitive
  apiGroup: rbac.authorization.k8s.io
```
We could then have something like this:
```
systemsUnderTest:
- kind: Deployment
  apiGroup: apps
  name: service-catalog 
  namespace: kyma-system
```

Reply:
@sjanota It is too granular IMO. We should work on the "feature gates" or helm charts level, not on the level of specific deployments.

Reply:
Additional cluster scoped CRD TestSystem? It will be empty (for now) with only it's name that matters.

Reply:
@sjanota We have BackendModules in kyma which probably we can use for it. But I don't want to have a strict dependency to Kyma and because of that, I propose strings.

Reply:
We will use LabelsSelector instead of compoents